core/src/main/java/com/linecorp/armeria/client/RequestOptionsSetters.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Request;
+import com.linecorp.armeria.common.Response;
+
+import io.netty.util.AttributeKey;
+
+interface RequestOptionsSetters {
```

Comment:
Could we also add `requestOptions(RequestOptions)` that overwrites the options? so that a user can do this in `WebClientRequestPreparation`?

```java
webClient.prepare()
         .requestOptions(myRequestOptions)
         ....
         .execute()
```

Reply:
How about adding `requestOptions(RequestOptions)` to `RequestOptionsSetters`?

Reply:
I also wonder what would be the right behavior of `requestOptions(RequestOptions)` regarding attributes. Should it clear the old attrs or not?

Reply:
Probably better not clearing the old ones just like we don't for similar properties in `ClientOptions`.

Reply:
> How about adding requestOptions(RequestOptions) to RequestOptionsSetters?

`WebClient.prepare().requestOptions(...)` sounds good. But `RequstOptions.builder().requestTimeout(...).requestOptions(...)` is a bit weird to me. How about leaving `requestOptions(RequestOptions)` in `WebClientRequestPreparation` and adding `copyOf(RequestOptions)` to   `RequestOptionsBuilder`?

> I also wonder what would be the right behavior of requestOptions(RequestOptions) regarding attributes. Should it clear the old attrs or not?

If it is ambiguous, how about adding two versions of setters?
```
setAttr(AttributeKey<V>, V)
setAttrIfAbsent(AttributeKey<V>, V)
```

Reply:
I see. How about just adding `RequestOptions.builder(RequestOptions)` that allows a user to source the initial values from another `RequestOptions`?

Reply:
> How about just adding RequestOptions.builder(RequestOptions)

That sounds good.